beacon-chain/rpc/apimiddleware/custom_handlers.go

```diff
@@ -193,17 +201,53 @@ func handlePostSSZ(
 	return true
 }
 
-func sszRequested(req *http.Request) bool {
+func sszRequested(req *http.Request) (bool, error) {
 	accept, ok := req.Header["Accept"]
 	if !ok {
-		return false
+		return false, nil
+	}
+	if len(accept) == 0 {
+		return false, nil
 	}
-	for _, v := range accept {
-		if v == "application/octet-stream" {
-			return true
+	types := strings.Split(accept[0], ",")
+	// match a number with optional decimals
+	regex, err := regexp.Compile("q=\\d(\\.\\d)?")
+	if err != nil {
+		return false, err
+	}
+	currentType, currentPriority := "", 1.0
+	for _, t := range types {
+		values := strings.Split(t, ";")
+		name := values[0]
+		if name != jsonMediaType && name != octetStreamMediaType {
+			continue
+		}
+		// no params specified
+		if len(values) == 1 {
+			if currentType == "" {
+				currentType = name
+				continue
+			}
+			priority := 1.0
+			if priority > currentPriority {
+				currentType, currentPriority = name, priority
+			}
+			continue
+		}
+		params := values[1]
+		match := regex.Find([]byte(params))
+		if match != nil {
+			priority, err := strconv.ParseFloat(strings.Split(string(match), "=")[1], 32)
```

Comment:
If you're going to take on the overhead of using a regex I would suggest simplifying things by using the `String` regexp methods and using a capture group to extract the desired value, via https://pkg.go.dev/regexp#Regexp.FindAllStringSubmatch
```
prioRx := regexp.MustCompile(`q=(\d+(?:\.\d+)?)`) // note that I added a `+` to the `\d` shorthands, otherwise you'll only capture the first character
pg := prioRx.FindAllStringSubmatch(params, 1)
if len(pg) != 1 {
    continue
}
priority, err := strconv.ParseFloat(pg[0][1])
...
```
more self-contained example: https://go.dev/play/p/NRswRIwv2Tr